ferveo/src/api.rs

```diff
@@ -640,104 +658,137 @@ mod test_ferveo_api {
+    #[test_case(7, 7; "number of shares (validators) is not a power of 2")]
+    #[test_case(4, 6; "number of validators greater than the number of shares")]
+    fn server_side_local_verification(shares_num: u32, validators_num: u32) {
         let rng = &mut StdRng::seed_from_u64(0);
+        let security_threshold = shares_num / 2 + 1;
 
-        let (messages, validators, _) =
-            make_test_inputs(rng, TAU, SECURITY_THRESHOLD, SHARES_NUM);
+        let (messages, validators, _) = make_test_inputs(
+            rng,
+            TAU,
+            security_threshold,
+            shares_num,
+            validators_num,
+        );
 
         // Now that every validator holds a dkg instance and a transcript for every other validator,
         // every validator can aggregate the transcripts
         let me = validators[0].clone();
         let mut dkg =
-            Dkg::new(TAU, SHARES_NUM, SECURITY_THRESHOLD, &validators, &me)
+            Dkg::new(TAU, shares_num, security_threshold, &validators, &me)
                 .unwrap();
 
-        let local_aggregate = dkg.aggregate_transcripts(&messages).unwrap();
-        assert!(local_aggregate
-            .verify(dkg.0.dkg_params.shares_num(), &messages)
-            .is_ok());
+        let good_aggregate = dkg.aggregate_transcripts(&messages).unwrap();
+        assert!(good_aggregate.verify(validators_num, &messages).is_ok());
 
         // Test negative cases
 
         // Notice that the dkg instance is mutable, so we need to get a fresh one
         // for every test case
 
+        // Should fail if the number of validators is less than the number of messages
+        assert!(good_aggregate
+            .verify(messages.len() as u32 - 1, &messages)
+            .is_err());
+
         // Should fail if no transcripts are provided
         let mut dkg =
-            Dkg::new(TAU, SHARES_NUM, SECURITY_THRESHOLD, &validators, &me)
+            Dkg::new(TAU, shares_num, security_threshold, &validators, &me)
                 .unwrap();
         let result = dkg.aggregate_transcripts(&[]);
         assert!(result.is_err());
 
         // Not enough transcripts
         let mut dkg =
-            Dkg::new(TAU, SHARES_NUM, SECURITY_THRESHOLD, &validators, &me)
+            Dkg::new(TAU, shares_num, security_threshold, &validators, &me)
                 .unwrap();
-        let not_enough_messages = &messages[..SECURITY_THRESHOLD as usize - 1];
-        assert!(not_enough_messages.len() < SECURITY_THRESHOLD as usize);
+        let not_enough_messages = &messages[..security_threshold as usize - 1];
+        assert!(not_enough_messages.len() < security_threshold as usize);
         let insufficient_aggregate =
             dkg.aggregate_transcripts(not_enough_messages).unwrap();
-        let result = insufficient_aggregate.verify(SHARES_NUM, &messages);
+        let result = insufficient_aggregate.verify(validators_num, &messages);
         assert!(result.is_err());
```

Comment:
I'm not sure this should fail only because the number of transcripts is insufficient. This test is probably expecting a failure due to an unrelated issue.